### PR TITLE
fix special FQ node

### PR DESCRIPTION
**Pull request by usstq**
### Details:
 - *FQ node with input_low==input_high was observed in many INT8 models, according to definition, it performs binarization instead of normal quantization, in this PR we trying to support this kind of special FQ node using normal procedure*
 - *...*

### Tickets:
 - *CVS-97942*
 - *CVS-98850*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@tomdol  @mateusztabaka  could you please take a look on ONNX FE changes?

**Review by tomdol — APPROVED** [review]
Overall the code is correct but I think it should rather be moved to some optimization transformation like NOP-elimination (or a new dedicated transform which replaces an obsolete FQ node). Please leave it as is and create a ticket to move the code to a transformation or address in this PR.
Also I think at least a single test for this corner case should be added. The ONNX quantized operators are tested here: src/frontends/onnx/tests/onnx_import_quant.in.cpp

File: src/plugins/intel_cpu/src/nodes/fake_quantize.cpp
Comment (zhangYiIntel) [inline src/plugins/intel_cpu/src/nodes/fake_quantize.cpp:1899]: Just Like the ONNX model, I think the special FQ in IR models also should be supported in LPT transformation which affects less in execution time.

**Comment by usstq** [comment]
@tomdol @zhangYiIntel @dmitry-gorokhov Hi all, I changed the solution, according to @tomdol's comment, instead of trying to support such special FQ node at runtime in plugin, I added 2 const folding passes to identify and replace these special FQ nodes with constant+broadcast subgraph which totally eliminates them from the runtime graph.

The root cause of such special FQ node is insufficient calibration dataset used for quantization of models using  POT/nncf tool. for example, when no sample in calibration dataset can produce positive activations in output of Relu/leakyRelu/PRelu layer, the POT/nncf tool will insert FQ with inputLow==inputHigh==outputLow==outputHigh==0, 

```
    f32_[1,1,1,1]  org_1989199316641 = Constant(0)      attrs: element_type=f32 shape=[1,1,1,1] value=?
    f32_[1,1,1,1]  org_1990199415288 = Constant(0)      attrs: element_type=f32 shape=[1,1,1,1] value=?
    f32_[1,1,1,1]  org_1991199516668 = Constant(0)      attrs: element_type=f32 shape=[1,1,1,1] value=?
    f32_[1,1,1,1]  org_1992199616785 = Constant(0)      attrs: element_type=f32 shape=[1,1,1,1] value=?
    f32_[1,64,8,8]  org_Conv_265/WithoutBiases/fq_input_0 = FakeQuantize(org_LeakyRelu_264,org_1989199316641,org_1990199415288,org_1991199516668,org_1992199616785)      attrs: levels=256 auto_broadcast=numpy
```

similarly pytorch/onnx quantization w/o finetuning also procedure quantization/dequantization node with scale equal to zero, and since DequantizeLinear node from onnx will be transformed into multiply(subtract(x, zero_point), scale), this creates pattern such as `y=multiply(x, 0)`

These kind of special FQ node causes trouble in runtime, since FQ with same inputLow & inputHigh requires totally different formula to implement, and more important, inference of such FQ actually is totally waste of time since such computation will finally be Dequantized back to zero,

Thus I introduced two more patterns in LPT's `FoldFakeQuantizeTransformation`:
 1. FQ node with same outputLow/outputHigh inputs: they will be replaced with const-subgraph of outputLow+broadcast;
 2. Multiple of data node (x) with constant all zero tensor (call it `a`): it will be replaced with const-subgraph of a+broadcast;
 
with these changes, 2 models in jira tickets can be inference on plugin w/o any change in plugin, I only tested on CPU, but I guess it should work on other plugins also, I also tested accuracy on 75 quantized models and no regression is found.

File: src/plugins/intel_cpu/src/nodes/fake_quantize.cpp
Comment (usstq) [reply src/plugins/intel_cpu/src/nodes/fake_quantize.cpp]: Thanks a lot, that's good idea, the fix is changed accordingly!

**Comment by dmitry-gorokhov** [comment]
@v-Golubev  @eshoguli  reassigning to you since now it is LPT related changes.

File: src/common/low_precision_transformations/src/fold_fake_quantize.cpp
Comment (usstq) [reply src/common/low_precision_transformations/src/fold_fake_quantize.cpp:41]: test case added!

File: src/common/low_precision_transformations/src/fold_fake_quantize.cpp
Comment (usstq) [reply src/common/low_precision_transformations/src/fold_fake_quantize.cpp:64]: the `dequantizeMultiply`  is actually a normal multiply, not necessarily generated by `FakeQuantizeDecompositionTransformation` , it may comes directly from onnx frontend when translate `DequantizeLinear` op, and it's not FQ, so it only has 2 inputs, right?

File: src/common/low_precision_transformations/tests/fake_quantize_transformation.cpp
Comment (usstq) [inline src/common/low_precision_transformations/tests/fake_quantize_transformation.cpp:186]: dot interval FQ is being const-folded in this PR, thus the old compareFunction test case is removed

File: src/common/low_precision_transformations/tests/fold_fake_quantize_in_transformations.cpp
Comment (usstq) [inline src/common/low_precision_transformations/tests/fold_fake_quantize_in_transformations.cpp:84]: to make sure the test hits the dot-interval const-fold case rather than generic FQ-const-fold, we must use parameter as data input rather than const

File: src/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/fake_quantize_transformation.cpp
Comment (usstq) [inline src/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/fake_quantize_transformation.cpp:51]: This test cannot success after FQ is being const-folded, because GPU plugin will also const-fold Pooling layer following FQ node, thus Pooling layer is gone.

File: src/common/low_precision_transformations/src/fold_fake_quantize.cpp
Comment (v-Golubev) [inline src/common/low_precision_transformations/src/fold_fake_quantize.cpp:71]: I suggest using `NetworkHelper::getConstantInput` here to get `constScales`

File: src/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/fake_quantize_transformation.cpp
Comment (v-Golubev) [reply src/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/fake_quantize_transformation.cpp:51]: Can we leave this test-case but don't check Pooling precision?

File: src/common/low_precision_transformations/src/fold_fake_quantize.cpp
Comment (v-Golubev) [inline src/common/low_precision_transformations/src/fold_fake_quantize.cpp:25]: We handle in this transformation completely different cases without any common parts: FakeQuantize on constant path or with equal constants, and Multiply with zero. I think this could be a good reason to move Multiply part to separate transformation (something like `FoldZeroMultiply`). @eshoguli what do you think?

File: src/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/fake_quantize_transformation.cpp
Comment (usstq) [reply src/tests/functional/plugin/gpu/shared_tests_instances/low_precision_transformations/fake_quantize_transformation.cpp:51]: OK

File: src/common/low_precision_transformations/src/fold_fake_quantize.cpp
Comment (usstq) [reply src/common/low_precision_transformations/src/fold_fake_quantize.cpp:100]: Great!

File: src/common/low_precision_transformations/src/fold_fake_quantize.cpp
Comment (usstq) [reply src/common/low_precision_transformations/src/fold_fake_quantize.cpp:71]: Sorry, but I also needs dataNode which we cannot deduce easily from `NetworkHelper::getConstantInput`'s return value, thus I prefer to leave it as it is

File: src/common/low_precision_transformations/src/fold_fake_quantize.cpp
Comment (usstq) [reply src/common/low_precision_transformations/src/fold_fake_quantize.cpp]: OK, separated

Comment (usstq) [reply src/common/low_precision_transformations/src/fold_fake_quantize.cpp]: Looks like we also need someone from `openvino-docs-maintainers` to review

**Comment by usstq** [comment]
@v-Golubev I've fixed the PR according to your comment, can you please review it again ? Thx!

**Review by eshoguli — CHANGES_REQUESTED** [review]
Offline discussion was started in 98850 ticket. Please, don't merge before the discussion is completed. The general questions:
1. Looks like OV works without changes. 
2. How the model can be working if inputs are not used?
3. Inputs are not removed from the model.
4. Looks like FoldZeroMultiplyTransformation is not used in the ticket model.

**Review by v-Golubev — COMMENTED** [review]
Just a few more comments.
Also, due to the fact that these changes might lead to unused model `Parameters`, I suggest validating model caching for the affected models. Thanks!

File: src/common/low_precision_transformations/src/fold_zero_multiply.cpp
Comment (usstq) [reply src/common/low_precision_transformations/src/fold_zero_multiply.cpp]: fold_zero_multiply.cpp was removed according to @eshoguli 's comment

File: src/common/low_precision_transformations/src/fold_zero_multiply.cpp
Comment (usstq) [reply src/common/low_precision_transformations/src/fold_zero_multiply.cpp]: fold_zero_multiply.cpp was removed according to @eshoguli 's comment

**Comment by usstq** [comment]
@eshoguli PR updated, fold_zero_multiply.cpp was removed according to your suggestion, please review again, Thx!

**Comment by wenjiew** [comment]
Can we speed up this PR? It is related to a customer model. Thanks!

**Review by v-Golubev — APPROVED** [review]
No comments left from my side.
But I still think that we need to check that models caching behavior isn't affected by these changes (at least, on the models from the tickets)
Also, we need to pass internal performance validation (if it hasn't been passed before) before merge.
Upd. agreed with @eshoguli that functional validation is enough. I made a run that didn't find any functional degradations

**Pull request merged**
